### PR TITLE
PR to update 2019-10

### PR DESCRIPTION
**Pull request by apeltzer**
PR to update to 2019/10, fix linting etc pp

## PR checklist
 - [x] This comment contains a description of changes (with reason)
 - [x] If you've fixed a bug or added code that should be tested, add tests!
 - [ ] If necessary, also make a PR on the [nf-core/ampliseq branch on the nf-core/test-datasets repo]( https://github.com/nf-core/test-datasets/pull/new/nf-core/ampliseq)
 - [ ] Ensure the test suite passes (`nextflow run . -profile test,docker`).
 - [ ] Make sure your code lints (`nf-core lint .`).
 - [ ] Documentation in `docs` is updated
 - [ ] `CHANGELOG.md` is updated
 - [ ] `README.md` is updated

**Learn more about contributing:** https://github.com/nf-core/ampliseq/tree/master/.github/CONTRIBUTING.md

### CODE REVIEW

File: docs/output.md [hunk 1]
```diff
@@ -56,7 +56,7 @@ The pipeline has special steps which allow the software versions used to be repo
 For more information about how to use MultiQC reports, see [http://multiqc.info](http://multiqc.info)
 
 ### QIIME2
-**Quantitative Insights Into Microbial Ecology 2** ([QIIME2](https://qiime2.org/)) is a next-generation microbiome bioinformatics platform and the successor of the widely used [QIIME1](https://www.nature.com/articles/nmeth.f.303). QIIME2 is currently **under heavy development** and often updated, this version of ampliseq uses QIIME2 2018.6. QIIME2 has a wide variety of analysis tools available and has excellent support in its [forum](https://docs.qiime2.org/2018.6/).
+**Quantitative Insights Into Microbial Ecology 2** ([QIIME2](https://qiime2.org/)) is a next-generation microbiome bioinformatics platform and the successor of the widely used [QIIME1](https://www.nature.com/articles/nmeth.f.303). QIIME2 is currently **under heavy development** and often updated, this version of ampliseq uses QIIME2 2019.7. QIIME2 has a wide variety of analysis tools available and has excellent support in its [forum](https://docs.qiime2.org/2019.7/).
```
Comment (d4straub) [inline docs/output.md]: Upgraded to 2019.10

File: docs/output.md [hunk 2]
```diff
@@ -205,15 +205,15 @@ ANCOM is applied to each suitable or specified metadata column for 6 taxonomic l
   * taxonomic level: level-2 (phylum), level-3 (class), level-4 (order), level-5 (family), level-6 (genus), ASV
 
 ## More help
-QIIME2 is currently **under heavy development** and often updated, this version of ampliseq uses QIIME2 2018.6. QIIME2 has excellent support in its [forum](https://docs.qiime2.org/2018.6/).
+QIIME2 is currently **under heavy development** and often updated, this version of ampliseq uses QIIME2 2019.7. QIIME2 has excellent support in its [forum](https://docs.qiime2.org/2019.7/).
```
Comment (d4straub) [inline docs/output.md]: 2019.10

File: docs/usage.md [hunk 3]
```diff
@@ -184,7 +184,7 @@ This is optional, but for performing downstream analysis such as barplots, diver
 Please note the following requirements:
 
 1. The path must be enclosed in quotes
-2. The metadata file has to follow the [QIIME2 specifications](https://docs.qiime2.org/2018.6/tutorials/metadata/)
+2. The metadata file has to follow the [QIIME2 specifications](https://docs.qiime2.org/2019.7/tutorials/metadata/)
```
Comment (d4straub) [inline docs/usage.md]: 2019.10

File: docs/usage.md [hunk 4]
```diff
@@ -358,9 +358,9 @@ If you have trained a compatible classifier before, from sources such as [SILVA]
 Please note the following requirements:
 
 1. The path must be enclosed in quotes
-2. The cassifier is a Naive Bayes classifier produced by "qiime feature-classifier fit-classifier-naive-bayes" (e.g. by this pipeline or from [QIIME2 resources](https://docs.qiime2.org/2018.6/data-resources/))
+2. The cassifier is a Naive Bayes classifier produced by "qiime feature-classifier fit-classifier-naive-bayes" (e.g. by this pipeline or from [QIIME2 resources](https://docs.qiime2.org/2019.7/data-resources/))
```
Comment (d4straub) [inline docs/usage.md]: 2019.10

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,13 @@
 # nf-core/ampliseq
 
+## nf-core/ampliseq version 1.1.1
+
+#### Pipeline updates
+* Update from QIIME2 v2018.6 to v2019.7, including DADA2 v1.6 to DADA2 v1.10
```
Comment (d4straub) [inline CHANGELOG.md]: Update from QIIME2 v2018.6 to **v2019.10**, including DADA2 v1.6 to DADA2 v1.10

**Comment by apeltzer** [comment]
This needs this call first: https://github.com/nf-core/ampliseq/pull/109

THat will update the docker container on dev, then we can aim at fixing remaining points in here to get a new 1.1.1 release out there 👍 @d4straub can you review the other PR too quickly?

**Comment by apeltzer** [comment]
Ok @d4straub -we can now adjust the remaining parameter changes. Once that is done, this should be good to go ...?

**Comment by d4straub** [comment]
The original pull request had all required changes for 2019.7, to my knowledge there shouldn't be any parameter changes required for 2019.10 coming from 2019.7. So the main.nf should work when using the 2019.10 container.

**Comment by apeltzer** [comment]
Waiting for the `dev` image to build, to commit my last changes (all your raised points @d4straub - linting works locally just fine :-). Then we can update the commands as per your comment... 

https://hub.docker.com/repository/registry-1.docker.io/nfcore/ampliseq/builds/cc24677f-f94f-4ad8-bb5c-4e64f2d1f2cd

**Comment by apeltzer** [comment]
I'll retrigger all tests once the Docker Hub image is then there - or you do it in like an hour ;-)

**Comment by MaxUlysse** [comment]
Looks good, but issues with tests

**Comment by apeltzer** [comment]
@d4straub 

```
and executed:

  qiime tools import 					--type 'SampleData[PairedEndSequencesWithQuality]' 					--input-path manifest.txt 					--output-path demux.qza 					--input-format PairedEndFastqManifestPhred33

Command exit status:
  2

Command output:
  (empty)

Command error:
  Error: no such option: --input-format
```

**Comment by d4straub** [comment]
How is that possible? Runs flawless when I do:
```
nextflow pull -r 2019-7 nf-core/ampliseq
nextflow run nf-core/ampliseq -r 2019-7 -profile test,singularity
```

**Comment by d4straub** [comment]
Well, almost worked. Other than that it didn't finish successful, I cannot see a difference between Job 251.1 and 251.2 (versions, datasets...). I would be grateful if someone could enlighten me whats the difference.

**Comment by apeltzer** [comment]
Probably the container wasn’t yet done on docker - which was Auto fixed over night 🥳🥳

**Pull request merged**